central/alert/datastore/internal/store/postgres/store.go

```diff
@@ -672,6 +671,12 @@ func Destroy(ctx context.Context, db *pgxpool.Pool) {
 	dropTableAlerts(ctx, db)
 }
 
+// NewTestStore returns a new Store instance for testing
+func NewTestStore(ctx context.Context, db *pgxpool.Pool, gormDB *gorm.DB) Store {
```

Comment:
Is this how we usually handle cases like this where we need the class set up slightly differently for test?  I ask because when I see things like this I always concerned about something that isn't related to testing using it.  I would probably prefer to see the call to ApplySchemaForTable in the test setup itself if that were possible.

Reply:
Ideally if we could have another for test lib in this package. But since destroy is already here, I would add it here. I have the first version with ApplySchemaForTable in tests but this way it is much cleaner.

Reply:
I see what you are saying about the Destroy being there as a precedent, but along those lines, couldn't we make a CreateGormTestSchema function to match the Destroy instead of creating a separate instantiation?  All New does is set the pool.  So couldn't the test code do something like 
```
s.store = New(ctx, s.pool)
gormDB := pgtest.OpenGormDB(s.T(), source)
s.CreateGormTestSchema(ctx, gormDB) // or something like that
```

Reply:
See the store are created at many different tests in different pkgs, not only the local store tests. And the table name needs to come from different sources, and many of them are not auto generated. I then kept asking myself why I would ask for the table name from others if the store already had it. I wont try to figure out one by one that is a disaster I wont go back.

Reply:
fair enough.